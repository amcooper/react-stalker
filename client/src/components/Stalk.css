.Stalk {
  background-color: #07889B;
  color: white;
}
