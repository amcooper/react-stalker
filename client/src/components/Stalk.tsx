/// <reference path="./interfaces.d.ts" />

import * as React from "react";
import "./Stalk.css";

function Stalk(props: IStalkProps) {
  return (
    <div id="Stalk">
      <p>
        <button onClick={() => props.onEdit(props.item.id)}>edit</button>
        <button onClick={() => props.onDelete(props.item.id)}>delete</button>
        {props.item.stalker} spotted {props.item.celebrity}
        {" on "}
        {props.item.date} in {props.item.location}: &quot;
        {props.item.comment}
        &quot;
      </p>
    </div>
  );
}

export default Stalk;
