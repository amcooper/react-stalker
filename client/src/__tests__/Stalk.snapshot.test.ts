import * as React from "react";
import Stalk from "../components/Stalk";
import { create } from "react-test-renderer";
import { stalkList } from "../fixtures/fixtures";

const testStalk = stalkList[0];

describe("Stalk component renders the stalk correctly", () => {
  it("renders correctly", () => {
    const rendered = create(React.createElement(Stalk, {item: testStalk}));

    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
