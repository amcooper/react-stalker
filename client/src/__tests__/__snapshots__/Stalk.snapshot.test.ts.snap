// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stalk component renders the stalk correctly renders correctly 1`] = `
<div
  id="Stalk"
>
  <p>
    <button
      onClick={[Function]}
    >
      edit
    </button>
    <button
      onClick={[Function]}
    >
      delete
    </button>
    Terrell Devereaux spotted Rosie Perez
         on Mon Apr 23 2018 20:00:00 GMT-0400 (Eastern Daylight Time) in Vinegar Hill: 
        "Holla"
  </p>
</div>
`;
